crates/configuration/src/shared/node.rs

```diff
@@ -52,27 +52,36 @@ impl From<(&str, &str)> for EnvVar {
 }
 
 /// A node configuration, with fine-grained configuration options.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct NodeConfig {
     name: String,
-    image: Option<Image>,
-    command: Option<Command>,
+    pub(crate) image: Option<Image>,
+    pub(crate) command: Option<Command>,
```

Comment:
If we use setter we don't need to expose the fields to the crate (set_image/set_command are defined below), we can go only setters or only public field to crate to make it uniform.

Reply:
Will remove the setters. We will keep the pub(crate)